modules/nodes-pods-vertical-autoscaler-tuning.adoc

```diff
@@ -20,7 +20,165 @@ You can perform the following tunings on the VPA components by editing the `Vert
 
 * To configure the VPA Operator to monitor only workloads that are being managed by a VPA CR, set the `memory-saver` parameter to `true` for the recommender component.
 
-The following example VPA controller CR sets the VPA API QPS and burts rates, configures the component pod resource requests, and sets `memory-saver` to `true` for the recommender:
+For guidelines on the resources and rate limits that you could set for each VPA component, the following tables provide baseline values, depending on the size of your cluster and other factors. 
+
+[IMPORTANT]
+====
+These values were derived by internal Red{nbsp}Hat testing on clusters that are not necessarily representative of the loads a real-world cluster would experience. You should test these values in a non-production cluster before configuring a production cluster. 
+====
+
+.Requests by containers in the cluster
+[cols="1,1,1,1,1,1,1,1,1,3,3"]
+|===
+| Component 2+| 1-500 containers 2+| 500-1000 containers 2+| 1000-2000 containers 2+| 2000-4000 containers 2+| 4000+ containers
+
+| 
+| *CPU*
+| *Memory*
+| *CPU*
+| *Memory*
+| *CPU*
+| *Memory*
+| *CPU*
+| *Memory*
+| *CPU*
+| *Memory*
+
+s| Admission 
+| 25m 
+| 50Mi
+| 25m 
+| 75Mi 
+| 40m 
+| 150Mi 
+| 75m 
+| 260Mi 
+| (0.03 x c)/2 + 10 ^[1]^
+| (0.1 x c)/2 + 50 ^[1]^
+
+s| Recommender
+| 25m
+| 100Mi
+| 50m
+| 160Mi
+| 75m
+| 275Mi
+| 120m
+| 420Mi
+| (0.05 x c)/2 + 50 ^[1]^
+| (0.15 x c)/2 + 120 ^[1]^
+
+s| Updater
+| 25m
+| 100Mi
+| 50m
+| 220Mi
+| 80m
+| 350Mi
+| 150m
+| 500Mi
+| (0.07 x c)/2 + 20 ^[1]^
+| (0.15 x c)/2 + 200 ^[1]^
+
+|===
+[.small]
+. `c` is the number of containers in the cluster.
+
+[NOTE]
+====
+It is recommended that you set the memory limit on your containers to double the recommended requests in the table. However, because CPU is a compressible resource, setting CPU limits for containers can throttle the VPA. As such, it is recommended that you do not set a CPU limit on your containers.
+====
+
+.Rate limits by VPAs in the cluster
+[cols="1,2,2,2,1,2,1,2,1"]
+|===
+| Component 2+| 1 - 150 VPAs 2+| 151 - 500 VPAs 2+| 501-2000 containers 2+| 2001-4000 containers
+
+| 
+| *QPS Limit* ^[1]^
+| *Burst* ^[2]^
+| *QPS Limit*
+| *Burst*
+| *QPS Limit*
+| *Burst*
+| *QPS Limit*
+| *Burst*
+
+s| Recommender
+| 5
+| 10
+| 30
+| 60
+| 60
+| 120
+| 120
+| 240
+
+s| Updater
+| 5
+| 10
+| 30
+| 60
+| 60
+| 120
+| 120
+| 240
+
+|===
+[.small]
+. QPS specifies the queries per second (QPS) limit when making requests to Kubernetes API server. The default is `5.0`. 
+. Burst specifies the burst limit when making requests to Kubernetes API server. The default is `10.0`.
+
+[NOTE]
+====
+* If you have more than 4000 VPAs in your cluster, note the following recommendations: 
+** It is recommended that you start performance tuning with the values in the table and slowly increase the values until you achieve the desired recommender and updater latency and performance. 
```

Comment:
I would also add that the reason we want to performance tune slowly is because increased QPS and limits could also affect the cluster health if too many API requests are being sent to the kube-api-server from the VPA components.

Reply:
> increased QPS and limits

Should this be _increased QPS and Burst_ ??

Reply:
Yes you are right, I mispoke, should be QPS and Burst limits.